packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/Metric.test.tsx

```diff
@@ -0,0 +1,234 @@
+import { within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import * as React from 'react';
+
+import { renderWithThemeAndHookFormContext } from 'src/utilities/testHelpers';
+
+import { Metric } from './Metric';
+
+import type { CreateAlertDefinitionForm } from '../types';
+import type { AvailableMetrics } from '@linode/api-v4';
+const mockData: AvailableMetrics[] = [
+  {
+    available_aggregate_functions: ['min', 'max', 'avg'],
+    dimensions: [
+      {
+        dimension_label: 'cpu',
+        label: 'CPU name',
+        values: [],
+      },
+      {
+        dimension_label: 'state',
+        label: 'State of CPU',
+        values: [
+          'user',
+          'system',
+          'idle',
+          'interrupt',
+          'nice',
+          'softirq',
+          'steal',
+          'wait',
+        ],
+      },
+      {
+        dimension_label: 'LINODE_ID',
+        label: 'Linode ID',
+        values: [],
+      },
+    ],
+    label: 'CPU utilization',
+    metric: 'system_cpu_utilization_percent',
+    metric_type: 'gauge',
+    scrape_interval: '2m',
+    unit: 'percent',
+  },
+];
+
+describe('Metric component tests', () => {
+  const user = userEvent.setup();
+  it('should render all the components and names', () => {
+    const container = renderWithThemeAndHookFormContext<CreateAlertDefinitionForm>(
+      {
+        component: (
+          <Metric
+            apiError={[false, false]}
+            data={mockData}
+            name={`rule_criteria.rules.${0}`}
+            onMetricDelete={vi.fn()}
+            showDeleteIcon={false}
+          />
+        ),
+        useFormOptions: {
+          defaultValues: {
+            serviceType: 'linode',
+          },
+        },
+      }
+    );
+    expect(container.getAllByLabelText('Data Field'));
+    expect(container.getAllByLabelText('Aggregation Type'));
+    expect(container.getAllByLabelText('Operator'));
```

Comment:
add assertion in this tests

Reply:
It is getAllByLabelText, assertions are failing

Reply:
We are getting received as response of `container.getAllByLabelText(..)`

Reply:
Yeah, I figured out the mistake. The container is failing for the assertions, using screen now to check the label assertions